### PR TITLE
Improve AniList Logging

### PR DESCRIPTION
**Pull request by BrutuZ**
### Motivation for changes:
Plugin would generate crash reports when the AniList API was down, caused by raising an exception while handling an exception

### Detailed changes:
- Convert all logger calls to f-strings
- Don't raise exceptions to avoid crash reports.

### Addressed issues:
```
CRITICAL task_queue                    BUG: Unhandled exception during task queue run loop.
Traceback (most recent call last):
  File "github/flexget/flexget/plugins/input/anilist.py", line 119, in on_task_input
    json={'query': req_query, 'variables': req_variables},
  File "github/flexget/lib/site-packages/requests/sessions.py", line 590, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "github/flexget/flexget/utils/requests.py", line 271, in request
    result.raise_for_status()
  File "github/flexget/lib/site-packages/requests/models.py", line 953, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://graphql.anilist.co/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "github/flexget/flexget/task_queue.py", line 47, in run
    self.current_task.execute()
  File "github/flexget/flexget/task.py", line 87, in wrapper
    return func(self, *args, **kw)
  File "github/flexget/flexget/task.py", line 727, in execute
    self._execute()
  File "github/flexget/flexget/task.py", line 696, in _execute
    self.__run_task_phase(phase)
  File "github/flexget/flexget/task.py", line 517, in __run_task_phase
    for e in response:
  File "github/flexget/flexget/utils/cached_input.py", line 231, in __iter__
    for item in self.iterable:
  File "github/flexget/flexget/plugins/input/anilist.py", line 123, in on_task_input
    raise plugin.PluginWarning(f'Error reading list - {e}')
flexget.plugin.PluginError: Error reading list - 403 Client Error: Forbidden for url: https://graphql.anilist.co/
```

### CODE REVIEW

**Comment by gazpachoking** [comment]
Hmm. PluginErrors aren't supposed to generate crash reports, something seems weird with that log.

**Comment by gazpachoking** [comment]
I don't get how a PluginError was generated from this line `logger.error(f'Error reading list - {e}')` which seems to be what the traceback is saying.

**Comment by BrutuZ** [comment]
It was changed TO a `logger.error`, it was `raise plugin.PluginError(f'Error reading list - {e}')` before

**Comment by gazpachoking** [comment]
Let me double check on how PluginErrors are handling for generator input plugins. I'm thinking this is supposed to be handled elegantly, but the generator caching is messing it up.

File: flexget/plugins/input/anilist.py [hunk 1]
```diff
@@ -147,14 +149,13 @@ def on_task_input(self, task, config):
                                 'https://relations.yuna.moe/api/ids',
                                 json={'anilist': anime.get('id')},
                             ).json()
+                            logger.debug(f'Additional IDs: {ids}')
                         except RequestException as e:
+                            logger(f'Couldn\'t fetch additional IDs: {e}')
```
Comment (gazpachoking) [inline flexget/plugins/input/anilist.py:154]: What level is this meant to be?

Comment (BrutuZ) [reply flexget/plugins/input/anilist.py:154]: INFO

**Pull request merged**